src/rust/iced-x86-py/src/instruction.rs

```diff
@@ -166,6 +169,31 @@ impl Instruction {
 		Self { instr: iced_x86::Instruction::default() }
 	}
 
+	/// Set the internal state with the given unpickled state.
+	///
+	/// Args:
+	///     state (Any): unpickled state
+	#[pyo3(text_signature = "($self, state, /)")]
+	fn __setstate__(&mut self, py: Python, state: PyObject) -> PyResult<()> {
+		match state.extract::<&PyBytes>(py) {
+			Ok(s) => {
+				self.instr = deserialize(s.as_bytes()).unwrap();
+				Ok(())
+			}
+			Err(e) => Err(e),
+		}
+	}
+
+	/// Get the unpickled state corresponding to the instruction.
+	///
+	/// Returns:
+	///     int: (``bytes``) The unpickled state
+	#[pyo3(text_signature = "($self, /)")]
+	fn __getstate__(&self, py: Python) -> PyResult<PyObject> {
+		let state = PyBytes::new(py, &serialize(&self.instr).unwrap()).to_object(py);
```

Comment:
Should also return an error, probably same ValueError as above?

Reply:
Not sure serialize could really fail here since it's serializing our own instruction. I've added the PyValueError anyway.